src/sentry/tasks/deletion.py

```diff
@@ -129,7 +139,14 @@ def delete_project(object_id, transaction_id=None, continuous=True, **kwargs):
         p.update(status=ProjectStatus.DELETION_IN_PROGRESS)
 
     # Immediately revoke keys
-    ProjectKey.objects.filter(project_id=object_id).delete()
+    project_keys = ProjectKey.objects.filter(project_id=object_id)
+    deleted_keys = [pk.id for pk in project_keys]
+    project_keys.delete()
```

Comment:
You've changed the semantic here.

On L143 you've exhausted the lazy `QuerySet`.

You'd be safer doing:

```python
project_keys = list(ProjectKey.objects.filter(project_id=object_id).values_list('id', flat=True))
ProjectKey.objects.filter(id__in=project_keys).delete()
for key_id in project_keys:
  ...
```